File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FProgressBar.java
package io.github.gofaith.faithdroid.FViews;

import android.widget.ProgressBar;

import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.UIController;

public class FProgressBar extends FView implements AttrSettable, AttrGettable {
    public ProgressBar v;
    public FProgressBar(UIController uiController){
        parentController = uiController;
        v = new ProgressBar(parentController.activity);
        view=v;
    }
    @Override
    public String getAttr(String attr) {
        String str = getUniversalAttr(attr);
        if (str != null) {
            return str;
        }
        switch (attr){
            // ------------------------------------------
        }
        return "";
    }

    @Override
    public void setAttr(String attr, String value) {
        if (setUniversalAttr(attr, value)) {
            return;
        }
        switch (attr) {
            // -------------------------------------------------------------------
        }
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FProgressBar.kt
package io.github.stevenzack.faithdroidapp.FViews

import android.widget.ProgressBar
import io.github.stevenzack.faithdroidapp.UI.AttrGettable
import io.github.stevenzack.faithdroidapp.UI.AttrSettable
import io.github.stevenzack.faithdroidapp.UI.FView
import io.github.stevenzack.faithdroidapp.UI.UIController


class FProgressBar(uiController: UIController) : FView(), AttrSettable, AttrGettable {
    var v: ProgressBar

    init {
        parentController = uiController
        v = ProgressBar(parentController!!.activity)
        view = v
    }

   override fun getAttr(attr: String): String {
        val str = getUniversalAttr(attr)
        if (str != null) {
            return str
        }
        when (attr) {

        }// ------------------------------------------
        return ""
    }

   override fun setAttr(attr: String, value: String) {
        if (setUniversalAttr(attr, value)) {
            return
        }
        when (attr) {

        }// -------------------------------------------------------------------
    }
}
